test/groovy/PiperExecuteBinTest.groovy

```diff
@@ -209,6 +215,45 @@ class PiperExecuteBinTest extends BasePiperTest {
         assertThat(sshKey, is(['sshKey']))
     }
 
+    @Test
+    void testPiperExecuteBinCredentialsSetAsEnum() {
+        shellCallRule.setReturnValue('./piper getConfig --contextConfig --stepMetadata \'.pipeline/tmp/metadata/test.yaml\'', '{"sshCredentialsId":"sshKey", "fileCredentialsId":"credFile", "tokenCredentialsId":"credToken", "textCredentialsId":"credSecretText", "credentialsId":"credUsernamePassword", "dockerImage":"my.Registry/my/image:latest"}')
+
+        List sshKey = []
+        helper.registerAllowedMethod("sshagent", [List, Closure], {s, c ->
+            sshKey = s
+            c()
+        })
+
+        List stepCredentials = [
+            [type: Credential.FILE, id: 'fileCredentialsId', env: ['PIPER_credFile']],
+            [type: Credential.TOKEN, id: 'tokenCredentialsId', env: ['PIPER_credToken']],
+            [type: Credential.SECRET_TEXT, id: 'textCredentialsId', env: ['PIPER_credSecret']],
+            [type: Credential.USERNAME_PASSWORD, id: 'credentialsId', env: ['PIPER_user', 'PIPER_password']],
+            [type: Credential.SSH, id: 'sshCredentialsId'],
+        ]
```

Comment:
It could be a good idea to make this test dynamic. Something like:
```groovy
List stepCredentials = [[type: Credential.USERNAME_PASSWORD, id: 'credentialsId', env: ['PIPER_user', 'PIPER_password']]]
for (type in Credential.values()) {
    if (type != Credential.USERNAME_PASSWORD) {
        stepCredentials.add([type: type, id: "${type}CredentialsId", env: ["PIPER_cred${type}"]]
    }
}
```

... I don't know, maybe not worth the effort, since someone adding another enum value will likely not forget to extend the switch case in piperExecuteBin. Feel free to "resolve" if you think this is bogus.

Reply:
I like it, as now the tests will fail if somebody adds another credential type to the mix. Done.

Reply:
> I like it, as now the tests will fail if somebody adds another credential type to the mix. Done.

But only if you also make the asserts (1..3) dynamic, no? Sorry for not mentioning that before.

Reply:
But I **want** the tests to fail if someone adds another credential type, so I shouldn't go dynamic on the asserts, right?

Reply:
Maybe I am missing something, but I think it works the other ways around. If someone adds a CredentialType, but it is not included in your asserts, then it may be forgotten to also adjust the switch-statement in piperExecuteBin. You would not assert that it is handled there.

Reply:
-> so it needs to be automatically included in an assert, by making the asserts dynamic as well.

Reply:
makes sense, yes. Could you have a look now? Thx

Reply:
@stippi2 Could you check this one more time?